src/Draco.Compiler.Tests/EndToEnd/CompilingCodeTests.cs

```diff
@@ -92,6 +92,29 @@ public void Power()
         }
     }
 
+    [Fact]
+    public void PowerDecimalNumbers()
+    {
+        var assembly = Compile("""
+            func power(n: float64, exponent: int32): float64 = {
+                var i = 1;
+                var result = n;
+                while (i < exponent){
+                    result *= n;
+                    i += 1;
+                }
+                result
+            };
+            """);
+
+        var inputs = new[] { (1.5, 3), (5.28, 2), (-2.5, 4), (3, 3), (8.847, 2) };
+        foreach (var (n, exp) in inputs)
+        {
+            var pow = Invoke<double>(assembly, "power", n, exp);
+            Assert.Equal(Math.Pow(n, exp), pow);
```

Comment:
Please make sure that xUnit compares floats with some delta, as we shouldn't compare floats for exact equality